### PR TITLE
feat: staging environment overhaul

### PR DESCRIPTION
**Pull request by colinnielsen**
## Checklist

- [x] Documentation updated
- [x] Tested code changes

[ticket](https://linear.app/metropolis/issue/MP-544/fix-oauth2-dev-environment)

### CODE REVIEW

File: .github/workflows/integrationTest.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
         working-directory: forge-simple
         env:
           RPC_URL: http://127.0.0.1:8545
-          NO_PREVIEW_SERVICE: true
+          NO_METAL_SERVICE: true
```
Comment (colinnielsen) [inline .github/workflows/integrationTest.yml:43]: Included quite a few namespace changes with this pr

File: src/utils/preview-service.ts [hunk 2]
```diff
@@ -19,8 +19,21 @@ export type ChainConfig = {
...
+ */
+export const pingMetalService = async () => {
+  await fetch(`${METAL_SERVICE_URL}`).catch(async e => {
+    logDebug(e);
+    await exit(
+      'Error! Cannot connect to Metal servers',
+      MODE === 'dev' ? 'Make sure metal-service is running on port 1234' : '',
+    );
+  });
+};
```
Comment (kevinweaver) [inline src/utils/preview-service.ts:33]: nice! it might also make sense to skip this fetch if NO_METAL_SERVICE is set.

File: src/constants.ts [hunk 3]
```diff
@@ -4,20 +4,111 @@ export const SUPPORTED_CHAINS = Object.values(Network) as Network[];
...
-  : IS_DEV
-  ? 'http://localhost:3000'
-  : 'https://metal.build';
+  const env = process.env;
+
+  const MODE: 'dev' | 'staging' | 'prod' = flagIsPresent('--staging')
+    ? 'staging'
+    : flagIsPresent('--dev') || (process.argv[1].endsWith('mdev') && !flagIsPresent('--prod'))
+    ? 'dev'
+    : 'prod';
```
Comment (kevinweaver) [inline src/constants.ts:58]: 🎏 I like these flags! not blocking, but we might also consider throwing if more than one is passed. Something like: 
```
function maxOneFlag(...flags: string[]): boolean {
  return flags.filter(flagIsPresent).length <= 1;
}

if (!maxOneFlag('--staging', '--prod', '--dev')) {
  throw new Error('Only one of --staging, --prod, or --dev should be passed');
}
```
not sure if that belongs here or in some initial run validation.

File: src/utils/auth.ts [hunk 4]
```diff
@@ -189,7 +186,7 @@ export const listenForAuthorizationCode = async ({
  * @throws if the token request fails
  */
 export const requestForIdToken = async (body: URLSearchParams) => {
-  const url = `https://${AUTH0ֹֹֹֹֹ_VANITY_URI}/oauth/token`;
+  const url = `${AUTH0ֹֹֹֹֹ_ISSUER}/oauth/token`;
```
Comment (kevinweaver) [inline src/utils/auth.ts:189]: i like that the `https://` is no longer mandatory

**Review by kevinweaver — APPROVED** [review]

File: src/utils/preview-service.ts (hunk 2 above)
Comment (colinnielsen) [reply src/utils/preview-service.ts:33]: that's a really great note

File: src/constants.ts (hunk 3 above)
Comment (colinnielsen) [reply src/constants.ts:58]: will do

File: src/utils/auth.ts (hunk 4 above)
Comment (colinnielsen) [reply src/utils/auth.ts:189]: me too. this came from a misunderstanding on my part with how auth0 worked

**Pull request merged**